hasura/metadata/databases/staging/tables/public_circles.yaml

```diff
@@ -79,25 +116,3 @@ select_permissions:
           id:
             _eq: X-Hasura-User-Id
   role: user
-update_permissions:
-- permission:
-    check: null
-    columns:
-    - alloc_text
-    - auto_opt_out
-    - default_opt_in
-    - discord_webhook
-    - logo
-    - min_vouches
-    - name
-    - nomination_days_limit
-    - only_giver_vouch
-    - team_sel_text
-    - team_selection
-    - token_name
-    - vouching
-    - vouching_text
-    filter:
-      id:
-        _in: X-Hasura-Admin-Circle-Ids
-  role: user
```

Comment:
Any reason this was removed?

Reply:
From my understanding, hasura create/update with data validation/business logic are triggered in action handlers using admin secret. So i removed the permission for the role, or is it still needed ?

Reply:
As much as possible, we will want to write graphql mutations to update state. Data validation and business logic can be done either in the front end or with constraints in postgres.

Reply:
i'm not sure how would it work if data validation and business logic is on the frontend, if someone fork the frontend and remove the business logic and data validation could they bypass it ? Constraints for DB usually is used for ensuring data integrity rather than data validation. Maybe you could try (in this case one of the things you will need to check is for vouching and close all the nominations if its off), idk whether it will work

Reply:
> we will want to write graphql mutations to update state

@exrhizo yes, and, a hasura graphql mutation can call a webhook. we will want to have the business logic behind the mutation, not in the frontend, so that the same API can be used by multiple clients. e.g. in the case of vouching for a nominee, this could be done through a Discord bot in the future. you want a client to be able to simply call e.g. `addVouch` rather than need to directly manipulate our schema.

> hasura create/update with data validation/business logic are triggered in action handlers using admin secret. So i removed the permission for the role, or is it still needed ?

@zashton in this case, at least some of these are still needed, since circle admins can change these values from the circle admin settings page. when there isn't complex business logic needed, we should just do a simple update directly.

for slightly more complex validations we can use [check constraints](https://www.dropbox.com/s/5q6qx81qrgntgkg/Screen%20Shot%202022-01-06%20at%2010.10.21%20AM.png?dl=0) ([docs](https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-CHECK-CONSTRAINTS)) in the hasura console.

Reply:
@levity which endpoints do you think can be a simple update with no business logic ?

Reply:
> @levity which endpoints do you think can be a simple update with no business logic ?

circle admin settings is the first one that comes to mind... oh, i see what you're saying now, changing `min_vouches` has an effect on existing nominees. hmm, in this case, maybe we can split that out into a separate action.

Reply:
shouldn't it be together in one action handler, otherwise if 1 action succeeds and the other fails there will be data integrity issues. How does the client parse the constraint errors that the DB throws and display a meaningful error message for the user ?

Reply:
In this case, IMO, it's not really a data integrity issue if min_vouches is changed after some nominee already exists. This is more of a product question. Someone might want to change the setting for the future but not affect current nominees. So having this in two steps is not a big deal IMO.

That said, if you still want to do it in one step, using a webhook, I'm ok with that too.